### ISSUE TITLE
MergeableSelector bug with referencing parent selectors

### ISSUE THREAD

**Issue by KODerFunk** [brigade/scss-lint#520]
```scss
.b-some {
    &__image {
      width: 100%;
    }
    &__image + &__buttons {
      margin-top: 14px;
    }
}
```
report: `[W] MergeableSelector: Merge rule '&__image + &__buttons' with rule on line 2`
But this is not mergeable :(

**Comment by sds** [comment]
Yes, you can nest the `+ &__buttons` portion under `&__image`.

```scss
.b-some {
  &__image {
    width: 100%;

    + &__buttons {
      margin-top: 14px;
    }
  }
}
```

Hope that helps.

**Comment by KODerFunk** [comment]
Your variant is bad. My code process:
```css
.b-some__image {
  width: 100%;
}
.b-some__image + .b-some__buttons {
  margin-top: 14px;
}
```
Your:
```css
.b-some__image {
  width: 100%;
}
+ .b-some__image__buttons {
  margin-top: 14px;
}
```
Reopen issue please)

**Comment by sds** [comment]
Hey @KODerFunk,

Forgive my typo&mdash;I left out an `&`. The following generates the same code:

```scss
.b-some {
  &__image {
    width: 100%;

    & + &__buttons {
      margin-top: 14px;
    }
  }
}
```

**Comment by sds** [comment]
Thanks for pointing out my error, @KODerFunk.

@geniou: you implemented `MergeableSelector`. Any thoughts on this particular issue?

**Comment by piyush-avantsoft** [comment]
```css
.x-grid-row {
        height: 87px !important;

        .x-grid-cell {
            padding: 12px 16px 12px 25px;
        }

        .x-grid-cell-inner {
            padding: 0;
            // height: 65px;
        }
    }

    .x-grid-row.banker {
        min-height: 104px !important;

        .x-grid-cell, .x-grid-cell-inner, .dashboard-pipeline-item {
            min-height: 104px !important;
        }
    }

**Comment by geniou** [comment]
@sds it looks like its a (valid) bug.. my first idea would be to check for multiple `&`and then not to report a possible merge.. or what do you think?

**Comment by geniou** [comment]
@piyush-avantsoft isn't it mergable like this?

```scss
.x-grid-row {
  height: 87px !important;

  .x-grid-cell {
    padding: 12px 16px 12px 25px;
  }

  .x-grid-cell-inner {
    padding: 0;
  }

  &.sample{
    min-height: 104px !important;

    .x-grid-cell, .x-grid-cell-inner, .sample {
      min-height: 104px !important;
    }
  }
}
```

**Comment by sds** [comment]
@geniou yes, perhaps we shouldn't report the selector as mergeable if it contains multiple parent references.

**Comment by piyush-avantsoft** [comment]
@sds  thanks for help

**Comment by sds** [comment]
Hey @piyush-avantsoft,

I can't answer that question for you, as it's dependent on your application. `scss-lint` recommends limting your nesting depth to 3 since it's usually a sign of unnecessary complexity. In the code you posted above, it looks like you have a number of classes nested within each other. Is that necessary? For example, you could perhaps extract `.x-grid-cell-cell` out of `.x-grid-row` to get:

```scss
.x-grid-row,
  .x-grid-row-alt {
    color: $white-font-color !important;

    &.x-grid-row,
    &.x-grid-row-over {
       ...
    }
}

.x-grid-cell {
   ...    
}
```

Again, it's hard for anyone to give advice without context into your application. If the nesting depth constraints seem too strict, you should either disable the linter or up the maximum allowed depth in your configuration.

**Comment by piyush-avantsoft** [comment]
@sds  thanks for feedback 

so if I reduce the level of classes then will it resolve my performance issue?
Because currently my css size is near about 10 lacks lines.

so if you have any suggestion to improve performance then kindly forward me.

**Comment by stereodenis** [comment]
@piyush-avantsoft you problem is out of the issue.

@sds, @geniou maybe it is better to delete piyush-avantsoft's comments to cleanup discussion?

**Comment by brauliobo** [comment]
this was fixed on #534

**Comment by sds** [comment]
Fixed in fa06a602a25bf1ef83273d47e21e2bb78274536d.

**Comment by rbhalla** [comment]
I seem to be still experiencing this bug.
ruby 2.0.0
sass 3.4.22
scss_lint 0.48.0

This is the two selectors it is saying is mergeable:

    .report-preview__item:not(.is-editable) {
        &.is-section + &.is-story {
            .report-preview__story {
                padding-top: 20px;
            }
        }

        &.is-section {
            .report-textarea {
                margin-top: 30px;
                margin-bottom: 20px;
            }
        }
    }
